safe_transaction_service/history/services/collectibles_service.py

```diff
@@ -160,55 +164,73 @@ def __init__(self, ethereum_client: EthereumClient, redis: Redis):
             maxsize=4096, ttl=60 * 30
         )  # 2 hours of caching
 
+    def get_redis_metadata_timeout_key(self, uri: str):
+        return f"metadata_timeout:{uri}"
+
     @cachedmethod(cache=operator.attrgetter("cache_uri_metadata"))
     @cache_memoize(
-        60 * 60 * 24,
+        60 * 60 * 24 * 2,
         prefix="collectibles-_retrieve_metadata_from_uri",
         cache_exceptions=(MetadataRetrievalException,),
     )  # 1 day
-    def _retrieve_metadata_from_uri(self, uri: str) -> Any:
+    def _retrieve_metadata_from_uri(self, uri: str, force_retry: bool = False) -> Any:
         """
         Get metadata from URI. Currently just ipfs/http/https is supported
 
         :param uri: Metadata URI, like http://example.org/token/3 or ipfs://<keccak256>
         :return: Metadata as a decoded json
         """
-        uri = ipfs_to_http(uri)
-
-        if not uri or not uri.startswith("http"):
-            raise MetadataRetrievalException(uri)
-
-        try:
-            logger.debug("Getting metadata for uri=%s", uri)
-            with requests.get(uri, timeout=15, stream=True) as response:
-                if not response.ok:
-                    logger.debug("Cannot get metadata for uri=%s", uri)
-                    raise MetadataRetrievalException(uri)
-
-                content_length = response.headers.get("content-length", 0)
-                content_type = response.headers.get("content-type", "")
-                if int(content_length) > self.METADATA_MAX_CONTENT_LENGTH:
-                    raise MetadataRetrievalException(
-                        f"Content-length={content_length} for uri={uri} is too big"
-                    )
-
-                if "application/json" not in content_type:
-                    raise MetadataRetrievalException(
-                        f"Content-type={content_type} for uri={uri} is not valid, "
-                        f'expected "application/json"'
-                    )
-
-                logger.debug("Got metadata for uri=%s", uri)
+        from ..tasks import chance_to_retrieve_metadata_from_uri
```

Comment:
Task name should end `_task`. Also, `chance_to_retrieve` does not look clear to me

Reply:
Got it